### PR TITLE
Monolith ufw

### PR DESCRIPTION
**Pull request by benrudolph**
@proteusvacuum @nickpell this only allows certain traffic for monoliths, so no one can see our entire database...

http://manage.dimagi.com/default.asp?225659

cc: @millerdev

### CODE REVIEW

File: ansible/roles/elasticsearch/templates/config/elasticsearch.yml.j2 [hunk 1]
```diff
@@ -9,11 +9,6 @@ node.name: "{{ elasticsearch_node_name }}"
 
 transport.tcp.port: {{ elasticsearch_tcp_port }}
 
-{% if elasticsearch_bind_to_localhost %}
-# don't expose ES to public
-network.bind_host: localhost
```
Comment (benrudolph) [inline ansible/roles/elasticsearch/templates/config/elasticsearch.yml.j2]: this was actually just wrong. `host` is set to an IP address not localhost and thus it made it impossible to even access elastic search. the reason it didn't surface sooner is that no one restarted ES when the change was made

Comment (benrudolph) [inline ansible/roles/elasticsearch/templates/config/elasticsearch.yml.j2]: by default `bind_host` == `host`

File: ansible/roles/ufw/tasks/proxy_ufw.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    direction: in
+    port: 22
+    proto: any
+  when: is_monolith
+
+- name: allow mosh traffic
+  ufw:
+    rule: allow
+    direction: in
+    port: 60003
```
Comment (proteusvacuum) [inline ansible/roles/ufw/tasks/proxy_ufw.yml]: I think mosh takes a range of ports: `60000:61000`

File: ansible/roles/ufw/tasks/proxy_ufw.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    rule: allow
+    direction: in
+    port: "{{ item }}"
+    proto: tcp
+  with_items:
+    - 443  # https
+    - 80   # http
+  when: is_monolith
+
+- name: allow ssh traffic
```
Comment (proteusvacuum) [inline ansible/roles/ufw/tasks/proxy_ufw.yml]: do these allow ssh and mosh when not connected to the vpn?

File: ansible/roles/ufw/tasks/proxy_ufw.yml (hunk 2 above)
Comment (benrudolph) [inline ansible/roles/ufw/tasks/proxy_ufw.yml]: ah good to know, thanks

File: ansible/roles/ufw/tasks/proxy_ufw.yml (hunk 3 above)
Comment (benrudolph) [inline ansible/roles/ufw/tasks/proxy_ufw.yml]: yes, no sure what to do about that though since neither swiss or zambia are behind a vpn

Comment (benrudolph) [inline ansible/roles/ufw/tasks/proxy_ufw.yml]: we need to disable password ssh...

Comment (proteusvacuum) [inline ansible/roles/ufw/tasks/proxy_ufw.yml]: Ah. Do we need to add a second factor on monoliths then? Like google-auth or something?
(probably for discussion later, but just flagging now)

Comment (benrudolph) [inline ansible/roles/ufw/tasks/proxy_ufw.yml]: good question, i don't think we've thought about it too hard. agree that it's a parallel discussion, this is still better than every port being open 😄

**Comment by proteusvacuum** [comment]
:+1: on :heavy_check_mark:

File: ansible/roles/ufw/tasks/proxy_ufw.yml (hunk 3 above)
Comment (millerdev) [inline ansible/roles/ufw/tasks/proxy_ufw.yml]: are there not multiple interfaces, one for the VPN and one public? can we not have ufw be more permissive on the VPN interface than on the public one?

File: ansible/roles/ufw/tasks/proxy_ufw.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    direction: in
+    port: 22
+    proto: any
+  when: is_monolith
+
+- name: allow mosh traffic
+  ufw:
+    rule: allow
+    direction: in
+    port: "60000:61000"
```
Comment (millerdev) [inline ansible/roles/ufw/tasks/proxy_ufw.yml]: This is a big port range to have open to the public. See comment above about public vs VPN interfaces. That seems ideal. (However, I also agree that this is better than what we had before.)

**Pull request merged**